extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunner.java

```diff
@@ -425,7 +425,9 @@ public void registerListener(TaskRunnerListener listener, Executor executor)
   public Collection<TaskRunnerWorkItem> getRunningTasks()
   {
     List<TaskRunnerWorkItem> result = new ArrayList<>();
-    for (Pod existingTask : client.listPeonPods(Sets.newHashSet(PeonPhase.RUNNING))) {
+    for (Job existingTask : client.listAllPeonJobs().stream()
+        .filter(job -> job.getStatus() != null && job.getStatus().getActive() != null && job.getStatus().getActive() > 0).collect(Collectors.toSet())
```

Comment:
Instead of defining logic for whether the job is active, succeeded or failed in various places.  Can you create a class with a method for each state.  I.E

```
JobStatus.isActive(Job job)
JobStatus.isSucceeded(Job job)
JobStatus.isFailed(Job job)
```

Reply:
Good idea.

Reply:
i added a class to do this, I think it may change slightly once https://github.com/apache/druid/pull/14028#pullrequestreview-1373136808 is merged so I'll have to publish a new version once thats merged